core/deployment/src/main/java/io/quarkus/runner/bootstrap/GenerateConfigTask.java

```diff
@@ -0,0 +1,161 @@
+import io.quarkus.builder.BuildResult;
+import io.quarkus.deployment.builditem.ArchiveRootBuildItem;
+import io.quarkus.deployment.builditem.ConfigDescriptionBuildItem;
+import io.quarkus.deployment.util.FileUtil;
+
+/**
+ * This phase generates an example configuration file
+ *
+ * @author Stuart Douglas
+ */
+public class GenerateConfigTask {
+
+    private static final Logger log = Logger.getLogger(GenerateConfigTask.class);
+
+    private final Path configFile;
+
+    public GenerateConfigTask(Path configFile) {
+        this.configFile = configFile;
+    }
+
+    public Path run(CuratedApplication application) {
+        //first lets look for some config, as it is not on the current class path
+        //and we need to load it to run the build process
+        try {
+            AugmentAction augmentAction = new AugmentAction(application, Collections.emptyList());
+            BuildResult buildResult = augmentAction.runCustomAction(new Consumer<BuildChainBuilder>() {
+                @Override
+                public void accept(BuildChainBuilder chainBuilder) {
+                    chainBuilder.addFinal(ConfigDescriptionBuildItem.class);
+                    chainBuilder.addInitial(ArchiveRootBuildItem.class);
+                }
+            }, new Consumer<BuildExecutionBuilder>() {
+                @Override
+                public void accept(BuildExecutionBuilder buildExecutionBuilder) {
+                    try {
+                        buildExecutionBuilder.produce(new ArchiveRootBuildItem(Files.createTempDirectory("empty")));
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            });
+
+            List<ConfigDescriptionBuildItem> descriptions = buildResult.consumeMulti(ConfigDescriptionBuildItem.class);
+            Collections.sort(descriptions);
+
+            String existing = "";
+            if (Files.exists(configFile)) {
+                try (InputStream in = new FileInputStream(configFile.toFile())) {
+                    existing = new String(FileUtil.readFileContents(in), StandardCharsets.UTF_8);
+                }
+            }
+
+            StringBuilder sb = new StringBuilder();
+            for (ConfigDescriptionBuildItem i : descriptions) {
+                //we don't want to add these if they already exist
+                //either in commended or uncommented form
+                if (existing.contains("\n" + i.getPropertyName() + "=") ||
+                        existing.contains("\n#" + i.getPropertyName() + "=")) {
+                    continue;
+                }
+
+                sb.append("\n#\n");
+                sb.append(formatDocs(i.getDocs()));
+                sb.append("\n#\n#");
+                sb.append(i.getPropertyName() + "=" + i.getDefaultValue());
+                sb.append("\n");
+            }
+
+            try (FileOutputStream out = new FileOutputStream(configFile.toFile(), true)) {
```

Comment:
Same here. Or it looks like alternatively you could do `Files.newBufferedWriter` and write directly there instead of `sb` using `newLine()`.